src/mlpack/bindings/go/print_type_doc_impl.hpp

```diff
@@ -0,0 +1,143 @@
+/**
+ * @file print_type_doc_impl.hpp
+ * @author Yashwant Singh
+ *
+ * Print documentation for a given type.
+ */
+#ifndef MLPACK_BINDINGS_GO_PRINT_TYPE_DOC_IMPL_HPP
+#define MLPACK_BINDINGS_GO_PRINT_TYPE_DOC_IMPL_HPP
+
+#include "print_type_doc.hpp"
+
+namespace mlpack {
+namespace bindings {
+namespace go {
+
+/**
+ * Return a string representing the command-line type of an option.
+ */
+template<typename T>
+std::string PrintTypeDoc(
+    const util::ParamData& data,
+    const typename boost::disable_if<arma::is_arma_type<T>>::type*,
+    const typename boost::disable_if<util::IsStdVector<T>>::type*,
+    const typename boost::disable_if<data::HasSerialize<T>>::type*,
+    const typename boost::disable_if<std::is_same<T,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type*)
+{
+  // A flag type.
+  if (std::is_same<T, bool>::value)
+  {
+    return "A boolean flag option (true or false).";
+  }
+  // An integer.
+  else if (std::is_same<T, int>::value)
+  {
+    return "An integer (i.e., \"1\").";
+  }
+  // A floating point value.
+  else if (std::is_same<T, double>::value)
+  {
+    return "A floating-point number (i.e., \"0.5\").";
+  }
+  // A string.
+  else if (std::is_same<T, std::string>::value)
+  {
+    return "A character string (i.e., \"hello\").";
+  }
+  // Not sure what it is...
+  else
+  {
+    throw std::invalid_argument("unknown parameter type " + data.cppType);
+  }
+}
+
+/**
+ * Return a string representing the command-line type of a vector.
+ */
+template<typename T>
+std::string PrintTypeDoc(
+    const util::ParamData& data,
+    const typename std::enable_if<util::IsStdVector<T>::value>::type*)
+{
+  if (std::is_same<T, std::vector<int>>::value)
+  {
+    return "A array of integers; i.e., []int{0, 1, 2}.";
+  }
+  else if (std::is_same<T, std::vector<std::string>>::value)
+  {
+    return "A array of strings; i.e., []string{\"hello\", \"goodbye\"}.";
+  }
+  else
+  {
+    throw std::invalid_argument("unknown vector type " + data.cppType);
+  }
+}
+
+/**
+ * Return a string representing the command-line type of a matrix option.
+ */
+template<typename T>
+std::string PrintTypeDoc(
+    const util::ParamData& data,
+    const typename std::enable_if<arma::is_arma_type<T>::value>::type*)
+{
+  if (std::is_same<typename T::elem_type, double>::value
+          || std::is_same<typename T::elem_type, size_t>::value)
+  {
+    if (T::is_col || T::is_row)
+    {
+      return "A 1-d arraylike containing data.  This can be a 2-d matrix where "
+          "one dimension has size 1, or it can also be a gonum matrix. If the"
```

Comment:
Updated.

Reply:
I think the same change should be made for the 2-d matrices.  It's currently:

```
      return "A 2-d arraylike containing data.  This can be a gonum matrix.  "
             "If the type is not already float64, it will be converted.";
```

and I think you could use the same idea as above to change this to `A 2-d gonum Matrix.  If the type is not already float64, it will be converted.`  (I'm assuming that the second sentence is true, but I'm not 100% sure.)

Reply:
Updated.